lib/src/configuration.dart

```diff
@@ -295,3 +317,16 @@ class RealmSchema extends Iterable<SchemaObject> {
   @override
   SchemaObject elementAt(int index) => _schema.elementAt(index);
 }
+
+/// The signature of a callback that will be invoked if a client reset error occurs for this [Realm].
+///
+/// Currently, Flexible sync only supports the Manual Recovery.
+class SyncClientResetErrorHandler {
```

Comment:
Can we make that class abstract and its constructor private? That way people will be forced to use `ManualSyncClientResetHandler` instead. The reasoning here is to have a clear distinction between the interface and the concrete implementors so that when we add more options, people won't default to the wrong one.

Reply:
It's not a wrong per se. If someone wants to use this implementation or implement or extends it they can do it. No need to be too restrictive here imo

Reply:
We should not allow extending this class. The reason is because we're going to use the inheritor type to determine the resync mode. Something like:

```dart
ResyncMode getMode() {
  if (clientResetHandler is ManualResetHandler) {
    return ResyncMode.Manual;	  
  }

  // these will be added soon-ish
  if (clientResetHandler is DiscardLocalHandler) {
    return ResyncMode.DiscardLocal;
  }

  if (clientResetHandler is RecoverLocalHandler) {
    return ResyncMode.RecoverLocal;
  }

  throw RealmException('unsupported client reset handler');
}
```

Reply:
We can not stop people from implementing a class in Dart. 

I don't see are reason for these restriction. We could still do what you say and still allow custom implementations. In such cases we will call the callback of their implementation only. In this case it will be a `CustomClientResetHandler` implementation. 
Actually in the case above I would rather have a getter that returns the type of `ResyncMode` on the  ClientResetClass. Thus we don't throw `unsupported errors` but can work with any implementation of ClientResetClass.

I think we are talking to much in anticipation of the future. Which can go only so far. Lets not put the carriage before the horse.
I will resolve this and we can handle it later when the times comes.